dali/operators/decoder/audio/audio_decoder.h

```diff
@@ -36,12 +37,15 @@ class AudioDecoderBase {
     return OpenImpl(encoded);
   }
 
+  AudioMetadata OpenFromFile(const std::string &filepath) {
+    Close();
+    return OpenFromFileImpl(filepath);
+  }
```

Comment:
How about leveraging amazing technique of overloading? ;)
```c++
 AudioMetadata Open(span<const char> encoded) {
    Close();
    return OpenImpl(encoded);
  }

  AudioMetadata Open(const std::string &filepath) {
    Close();
    return OpenImpl(filepath);
  }
```

Reply:
I think it is clearer like this. Because a string could also be an encoded stream. It wouldn't be clear if we don't specify the name

Reply:
True